File: app/src/main/java/dev/ragnarok/fenrir/model/criteria/PhotoCriteria.java
package dev.ragnarok.fenrir.model.criteria;

import dev.ragnarok.fenrir.db.DatabaseIdRange;

public class PhotoCriteria {

    private final int accountId;
    private Integer ownerId;
    private Integer albumId;
    private String orderBy;
    private boolean sortInvert;

    private DatabaseIdRange range;

    public PhotoCriteria(int accountId) {
        this.accountId = accountId;
    }

    public DatabaseIdRange getRange() {
        return range;
    }

    public PhotoCriteria setRange(DatabaseIdRange range) {
        this.range = range;
        return this;
    }

    public boolean getSortInvert() {
        return sortInvert;
    }

    public PhotoCriteria setSortInvert(boolean invert) {
        sortInvert = invert;
        return this;
    }

    public int getAccountId() {
        return accountId;
    }

    public Integer getOwnerId() {
        return ownerId;
    }

    public PhotoCriteria setOwnerId(Integer ownerId) {
        this.ownerId = ownerId;
        return this;
    }

    public Integer getAlbumId() {
        return albumId;
    }

    public PhotoCriteria setAlbumId(Integer albumId) {
        this.albumId = albumId;
        return this;
    }

    public String getOrderBy() {
        return orderBy;
    }

    public PhotoCriteria setOrderBy(String orderBy) {
        this.orderBy = orderBy;
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/model/criteria/PhotoCriteria.kt
package dev.ragnarok.fenrir.model.criteria

import dev.ragnarok.fenrir.db.DatabaseIdRange

class PhotoCriteria(val accountId: Long) {
    var ownerId = 0L
        private set
    var albumId = 0
        private set
    var orderBy: String? = null
        private set
    var sortInvert = false
        private set
    var range: DatabaseIdRange? = null
        private set

    fun setRange(range: DatabaseIdRange?): PhotoCriteria {
        this.range = range
        return this
    }

    fun setSortInvert(invert: Boolean): PhotoCriteria {
        sortInvert = invert
        return this
    }

    fun setOwnerId(ownerId: Long): PhotoCriteria {
        this.ownerId = ownerId
        return this
    }

    fun setAlbumId(albumId: Int): PhotoCriteria {
        this.albumId = albumId
        return this
    }

    fun setOrderBy(orderBy: String?): PhotoCriteria {
        this.orderBy = orderBy
        return this
    }
}